frontend/apps/home/src/app/app.component.ts

```diff
@@ -1,22 +1,63 @@
-import {Component} from '@angular/core';
+import {Component, OnInit} from '@angular/core';
+import {ViewState} from './navbar/view-switch';
+import {animate, style, transition, trigger} from "@angular/animations";
 
 @Component({
   selector: 'hmd-root',
   template: `
-    <navbar (filterValueEmitter)="filterValue($event)"></navbar>
-    <libraries-list [filterValue]="filter">
-    </libraries-list>
+      <navbar
+              (filterValueEmitter)="filterValue($event)"
+              (viewStateEmitter)="changeView($event)"
+              [currentViewState]="currentViewState">
+      </navbar>
+      <libraries-list
+              *ngIf="isSelectedView(ViewState.CARDS)"
+              [filterValue]="filter"
+              [@fadeIn]>
+      </libraries-list>
+      <grid-view
+              *ngIf="isSelectedView(ViewState.GRID)"
+              [@fadeIn]>
+      </grid-view>
   `,
   styleUrls: [
     'app.component.scss'
+  ],
+  animations: [
+    trigger(
+      'fadeIn',
+      [
+        transition(':enter', [
+          style({opacity: 0}),
+          animate('0.5s ease-in')
+        ])
+      ]
+    )
   ]
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
+  filter = '';
+  ViewState = ViewState;
+  currentViewState: ViewState;
 
-  filter: string = '';
+  ngOnInit(): void {
+    let viewMode = localStorage.getItem('viewMode');
```

Comment:
zaraz mnie pogonisz, ale ja bym wydzielil to do serwisu - zapis viewMode i odczyt